frontend/public/module/utils/pod-utils.js

```diff
@@ -0,0 +1,146 @@
+import * as _ from 'lodash-es';
+import * as moment from 'moment';
+
+
+const POD_PHASES = ["Running", "Not Ready", "Warning", "Error", "Pulling", "Pending", "Succeeded", "Terminating", "Unknown"];
+
+const isPodTerminating = pod => {
+  return _.has(pod, 'metadata.deletionTimestamp');
+};
+
+const isPullingImage = pod => {
+  if (!pod) {
+    return false;
+  }
+
+  const phase = _.get(pod, 'status.phase');
+  if (phase !== 'Pending') {
+    return false;
+  }
+
+  const containerStatuses = _.get(pod, 'status.containerStatuses');
+  if (!containerStatuses) {
+    return false;
+  }
+
+  const containerPulling = function(containerStatus) {
+    // TODO: Update to use the pulling reason when available. We assume
+    // ContainerCreating === pulling, which might not be true.
+    return _.get(containerStatus, 'state.waiting.reason') === 'ContainerCreating';
+  };
+
+  return _.some(containerStatuses, containerPulling);
+};
+
+const isContainerLooping = containerStatus => {
+  return containerStatus.state.waiting && containerStatus.state.waiting.reason === 'CrashLoopBackOff';
+};
+
+const isContainerFailed = containerStatus => {
+  return containerStatus.state.terminated && containerStatus.state.terminated.exitCode !== 0;
+};
+
+const isContainerUnprepared = containerStatus => {
+  if (!containerStatus.state.running ||
+    containerStatus.ready !== false ||
+    !containerStatus.state.running.startedAt) {
+    return false;
+  }
+
+  // If this logic ever changes, update the message in podWarnings
+  const fiveMinutesAgo = moment().subtract(5, 'm');
+  const started = moment(containerStatus.state.running.startedAt);
+  return started.isBefore(fiveMinutesAgo);
+};
+
+const podWarnings = pod => {
```

Comment:
We should revisit all this logic from OpenShift console. This was written before a lot of APIs around conditions and container state were available to us.